toutv/client.py

```diff
@@ -140,17 +140,21 @@ def search(self, query):
         query_upper = query.upper()
         for emid, emission in emissions.items():
             if query_upper in emission.get_title().upper():
-                sr = toutv.bos.SearchResultData()
-                sr.Emission = emission
-                search.Results.append(sr)
+                try:
+                    # Load this emission' episodes, and add those to the search results too
+                    episodes = self._transport.get_emission_episodes(emission, True)
 
-                # Load this emission' episodes, and add those to the search results too
-                episodes = self._transport.get_emission_episodes(emission, True)
-                for epid, episode in episodes.items():
                     sr = toutv.bos.SearchResultData()
-                    sr.Episode = episode
+                    sr.Emission = emission
                     search.Results.append(sr)
 
+                    for epid, episode in episodes.items():
+                        sr = toutv.bos.SearchResultData()
+                        sr.Episode = episode
+                        search.Results.append(sr)
+                except Exception as e:
```

Comment:
I don't think we should catch all exceptions like this.  For example, we could have a bug in our code (e.g. reference an unexisting variable and it will go unnoticed).  We should catch the kind of exceptions that we know we can recover from.  And we should at least log the event (define a logger like in some of the other classes, and then `self._logger.error(...)`).